data_diff/__main__.py

```diff
@@ -1,129 +1,138 @@
-    db2 = connect_to_uri(db2_uri, threads)
-
-    if interactive:
-        db1.enable_interactive()
-        db2.enable_interactive()
-
-    start = time.time()
-
-    try:
-        options = dict(
-            min_time=min_age and parse_time_before_now(min_age), max_time=max_age and parse_time_before_now(max_age)
-        )
-    except ParseError as e:
-        logging.error("Error while parsing age expression: %s" % e)
-        return
-
-    table1 = TableSegment(db1, (table1_name,), key_column, update_column, columns, **options)
-    table2 = TableSegment(db2, (table2_name,), key_column, update_column, columns, **options)
-
-    differ = TableDiffer(
-        bisection_factor=bisection_factor,
-        bisection_threshold=bisection_threshold,
-        debug=debug,
-        threaded=threads != 1,
-        max_threadpool_size=threads,
-    )
-    diff_iter = differ.diff_tables(table1, table2)
-
-    if limit:
-        diff_iter = islice(diff_iter, int(limit))
-
-    if stats:
-        diff = list(diff_iter)
+from multiprocessing.sharedctypes import Value
+import sys
+import time
+import logging
+from itertools import islice
+
+from .diff_tables import TableSegment, TableDiffer
+from .database import connect_to_uri
+from .parse_time import parse_time_before_now, UNITS_STR, ParseError
+
+import rich
+import click
+
+LOG_FORMAT = "[%(asctime)s] %(levelname)s - %(message)s"
+DATE_FORMAT = "%H:%M:%S"
+
+COLOR_SCHEME = {
+    "+": "green",
+    "-": "red",
+}
+
+
+@click.command()
+@click.argument("db1_uri")
+@click.argument("table1_name")
+@click.argument("db2_uri")
+@click.argument("table2_name")
+@click.option("-k", "--key-column", default="id", help="Name of primary key column")
+@click.option("-t", "--update-column", default=None, help="Name of updated_at/last_updated column")
+@click.option("-c", "--columns", default=[], multiple=True, help="Names of extra columns to compare")
+@click.option("-l", "--limit", default=None, help="Maximum number of differences to find")
+@click.option("--bisection-factor", default=32, help="Segments per iteration")
+@click.option("--bisection-threshold", default=1024**2, help="Minimal bisection threshold")
+@click.option(
+    "--min-age",
+    default=None,
+    help="Considers only rows older than specified. "
+    "Example: --min-age=5min ignores rows from the last 5 minutes. "
+    f"\nValid units: {UNITS_STR}",
+)
+@click.option("--max-age", default=None, help="Considers only rows younger than specified. See --min-age.")
+@click.option("-s", "--stats", is_flag=True, help="Print stats instead of a detailed diff")
+@click.option("-d", "--debug", is_flag=True, help="Print debug info")
+@click.option("-v", "--verbose", is_flag=True, help="Print extra info")
+@click.option("-i", "--interactive", is_flag=True, help="Confirm queries, implies --debug")
+@click.option(
+    "-j", "--threads", default=None, help="Number of threads to use. 1 means no threading. Auto if not specified."
```

Comment:
Isn't the default 1 thread? What does auto mean? I think it should be 1 so that the output is nice and expected in the default case. It's also far safer on a relational database where much more could really have an impact (especially with the current queries).

Sorry I know this is not strictly related here, but I must've missed this on the other PR. I thought I read for this when I reviewed the threading PR

Reply:
Let's discuss it on slack.